src/plugins/testsBundle/index.js

```diff
@@ -18,21 +19,34 @@ module.exports = (kibana) => {
         let modules = [];
         let config = kibana.config;
 
-        // add the modules from all of the apps
-        for (let app of apps) {
-          modules = union(modules, app.getModules());
-        }
+        const testGlobs = ['src/ui/public/**/*.js'];
+        const testingPluginId = config.get('testsBundle.pluginId');
+
+        if (testingPluginId) {
+          const plugin = plugins.byId[testingPluginId];
+
+          // add the modules from all of this plugins apps
+          for (let app of plugin.apps) {
+            modules = union(modules, app.getModules());
+          }
 
-        const testGlobs = [
-          'src/ui/public/**/__tests__/**/*.js',
-        ];
+          testGlobs.push(
+            '!src/ui/public/**/__tests__/**/*',
+            `${plugin.publicDir}/**/__tests__/**/*.js`
+          );
+        } else {
 
-        for (const plugin of plugins) {
-          testGlobs.push(`${plugin.publicDir}/**/__tests__/**/*.js`);
+          // add the modules from all of the apps
+          for (let app of apps) {
+            modules = union(modules, app.getModules());
+          }
+
+          for (const plugin of plugins) {
```

Comment:
what? it works in chrome without transpilation... Can you link to part of the spec you're referring to?

![image](https://cloud.githubusercontent.com/assets/1329312/13011431/fe4dae66-d163-11e5-8349-12fd015054c6.png)